### PR TITLE
[tlv] Cleanup TLV module naming (remove CHIP from filenames).

### PR DESCRIPTION
**Pull request by turon**
Clean up naming of TLV files and headers by renaming `CHIPTLV` to just `TLV`.

Added soft links to old header names to ease transition by dependent code bases.

### CODE REVIEW

File: CHIPTLV.h [hunk 1]
```diff
@@ -0,0 +1 @@
+TLV.h
```
Comment (Damian-Nordic) [inline CHIPTLV.h:1]: Is it worth keeping these? Do we expect many external components include Matter's TLV headers directly?

**Review by Damian-Nordic — APPROVED** [review]

File: CHIPTLV.h (hunk 1 above)
Comment (andy31415) [reply CHIPTLV.h:1]: These feel odd to me as well ... are these expected to be here? They look like top level symlinks to me ... they feel odd.

**Review by andy31415 — DISMISSED** [review]
Remove the top level symlinks ... they seem added there by mistake (otherwise they would be under some "include" path)

File: CHIPTLV.h (hunk 1 above)
Comment (turon) [reply CHIPTLV.h:1]: Yes, they are wrong.  I'll remove.

**Comment by turon** [comment]
> Remove the top level symlinks ... they seem added there by mistake (otherwise they would be under some "include" path)

Done

**Review by andy31415 — APPROVED** [review]

**Pull request merged**